autorag/nodes/passagereranker/monot5.py

```diff
@@ -0,0 +1,95 @@
+    device = torch.device("cuda" if torch.cuda.is_available() else "cpu")
+    # Run async mono_t5_rerank_pure function
+    tasks = [mono_t5_pure(query, contents, scores, ids, model, device, tokenizer, token_false_id, token_true_id) \
+             for query, contents, scores, ids in zip(queries, contents_list, scores_list, ids_list)]
+    loop = asyncio.get_event_loop()
+    results = loop.run_until_complete(asyncio.gather(*tasks))
+    return results
+
+
+async def mono_t5_pure(query: str, contents: List[str], scores: List[float], ids: List[UUID],
+                       model, device, tokenizer, token_false_id, token_true_id) -> Tuple[List[str]]:
+    """
+    Rerank a list of contents based on their relevance to a query using MonoT5.
+    :param query: The query to use for reranking
+    :param contents: The list of contents to rerank
+    :param scores: The list of scores retrieved from the initial ranking
+    :param ids: The list of ids retrieved from the initial ranking
+    :param model: The MonoT5 model to use for reranking
+    :param device: The device to run the model on (GPU if available, otherwise CPU)
+    :param tokenizer: The tokenizer to use for the model
+    :param token_false_id: The id of the token used by the model to represent a false prediction
+    :param token_true_id: The id of the token used by the model to represent a true prediction
+    :return: tuple of lists containing the reranked contents, ids, and scores
+    """
+    model.to(device)
+
+    # Format the input for the model by combining each content with the query
+    input_texts = [f'Query: {query} Document: {content}' for content in contents]
+    # Tokenize the input texts and prepare for model input
+    input_encodings = tokenizer(input_texts, padding=True, truncation=True, max_length=512, return_tensors='pt').to(
+        device)
+
+    # Generate model predictions without updating model weights
+    with torch.no_grad():
+        outputs = model.generate(input_ids=input_encodings['input_ids'],
+                                 attention_mask=input_encodings['attention_mask'],
+                                 output_scores=True,
+                                 return_dict_in_generate=True)
+
+    # Extract logits for the 'false' and 'true' tokens from the model's output
+    logits = outputs.scores[-1][:, [token_false_id, token_true_id]]
+    # Calculate the softmax probability of the 'true' token
+    probs = torch.nn.functional.softmax(logits, dim=-1)[:, 1]  # Get the probability of the 'true' token
+
+    # Create a list of tuples pairing each content with its relevance probability
+    content_ids_probs = list(zip(contents, ids, probs.tolist()))
+
+    # Sort the list of pairs based on the relevance score in descending order
+    sorted_content_ids_probs = sorted(content_ids_probs, key=lambda x: x[2], reverse=True)
+
+    return tuple(map(list, zip(*sorted_content_ids_probs)))
```

Comment:
return Tuple[List[str], List[str], List[float]], which is (contents, ids, scores) as you said in annotation.